doc/guides/arch/arm_cortex_m.rst

```diff
@@ -0,0 +1,643 @@
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+| Memory protection               |                                   |                                                                                             |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   User mode                       |        N        |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   HW stack protection (MPU)       |        N        |   N     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   HW-assisted stack checking      |        N        |   N     |   N    |    N      |    N   |Y [#f2]_ |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+| HW-assisted null-pointer        |                                   |                 |         |        |           |        |         |            |            |
+| dereference detection           |                                   |        N        |   N     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+| HW-assisted atomic operations   |                                   |        N        |   N     |   Y    |    Y      |    Y   |    N    |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+|Support for non-cacheable regions|                                   |        N        |   N     |   Y    |    Y      |    Y   |    N    |     Y      |   Y        |
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
+| Execute from SRAM               |                                   |        N        |   N     |   Y    |    Y      |    Y   |    N    |     Y      |   Y        |
```

Comment:
The Cortex-M1 normally executes code from SRAM/Block RAM within an FPGA. Perhaps another footnote is needed?

Reply:
Right, this is more of a feature on XIP images to execute from SRAM, when code is normally in flash. I'll rephrase this, thanks.

Reply:
fixed.